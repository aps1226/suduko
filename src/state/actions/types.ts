export const SET_BOARD: string = 'SET_BOARD';

export const CHANGE_COLOR: string = 'CHANGE_COLOR';

export const SET_SELECTION: string = 'SET_SELECTION';