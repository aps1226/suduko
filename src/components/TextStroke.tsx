import * as React from "react";
import { StyleSheet, View } from "react-native";
import { Children, cloneElement, isValidElement } from "react";

type Props = {
  children: any,
  color: string,
  stroke: number
}
const styles = StyleSheet.create({
  outline: {
    position: 'absolute'
  },
});

export default function TextStroke(props:Props){
  const createClones = (w: number, h: number, color?: string) => {
    const { children } = props;
    return Children.map(children, child => {
      const currentProps = child.props as any;
      const currentStyle = (currentProps.style || {});
      const newProps = {
        ...currentProps,
        style: {
          ...currentStyle,
          textShadowOffset: {
            width: w,
            height: h
          },
          textShadowColor: color,
          textShadowRadius: 1
        }
      }
      return cloneElement(child, newProps)
    });
  }

  const {color, stroke, children} = props;
  const strokeW = stroke;
  const top = createClones(0, -strokeW * 1.2, color);
  const topLeft = createClones(-strokeW, -strokeW, color);
  const topRight = createClones(strokeW, -strokeW, color);
  const right = createClones(strokeW, 0, color);
  const bottom = createClones(0, strokeW, color);
  const bottomLeft = createClones(-strokeW, strokeW, color);
  const bottomRight = createClones(strokeW, strokeW, color);
  const left = createClones(-strokeW * 1.2, 0, color);

  return (
    <View>
      <View style={ styles.outline }>{ left }</View>
      <View style={ styles.outline }>{ right }</View>
      <View style={ styles.outline }>{ bottom }</View>
      <View style={ styles.outline }>{ top }</View>
      <View style={ styles.outline }>{ topLeft }</View>
      <View style={ styles.outline }>{ topRight }</View>
      <View style={ styles.outline }>{ bottomLeft }</View>
      { bottomRight }
    </View>
  );
  }
