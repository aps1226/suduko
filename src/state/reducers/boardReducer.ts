import { AnyAction } from 'redux';
import { SET_BOARD } from '../actions/types';
import { shuffle } from '../boardController';

const initBoard:(number | null)[][] = [shuffle([1,2,3,4,5,6,7,8,9]),[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null]];

const boardReducer = (state = initBoard, action: AnyAction):(number | null)[][]  =>{
  switch(action.type){
    case SET_BOARD:
      return action.payload;
    default:
      return state;
  }
}

export default boardReducer;