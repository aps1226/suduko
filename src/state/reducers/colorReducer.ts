import {CHANGE_COLOR,DEFAULT_COLORS} from '../actions/types';

import {Colors} from '../../../types'

const initGradient = ():string[] =>{
  const woodPalate:string[] = ["#a37e5c", "#b89372","#c8a484","#dcb999", "#d3af8f","#d9b696","#ceaa89","#d2ae8d","#b38e6d","aa8563"];
  const gradient:string[] = [];
  for(let i = 0; i < 100; i++){
    gradient.push(woodPalate[Math.round(Math.random()*10)])
  }
  return gradient;
}

const initializedGradient:string[] = initGradient();

const initialState:Colors = {
  1:['transparent','black'],
  2:['transparent','black'],
  3:['transparent','black'],
  4:['transparent','black'],
  5:['transparent','black'],
  6:['transparent','black'],
  7:['transparent','black'],
  8:['transparent','black'],
  9:['transparent','black'],
  'Selection_1':'transparent',
  'Selection_2':'transparent',
  'Selection_3':'transparent',
  'Selection_4':'transparent',
  'Selection_5':'transparent',
  'Selection_6':'transparent',
  'Selection_7':'transparent',
  'Selection_8':'transparent',
  'Selection_9':'transparent',
  'gradient': initializedGradient
}
const colorReducer = (state = initialState, action:any) =>{

  const initialState:Colors = {
    1:['transparent','black'],
    2:['transparent','black'],
    3:['transparent','black'],
    4:['transparent','black'],
    5:['transparent','black'],
    6:['transparent','black'],
    7:['transparent','black'],
    8:['transparent','black'],
    9:['transparent','black'],
    'Selection_1':'transparent',
    'Selection_2':'transparent',
    'Selection_3':'transparent',
    'Selection_4':'transparent',
    'Selection_5':'transparent',
    'Selection_6':'transparent',
    'Selection_7':'transparent',
    'Selection_8':'transparent',
    'Selection_9':'transparent',
    'gradient': initializedGradient
  }
  
  switch(action.type){
    case CHANGE_COLOR:
      if(!action.payload.includes('Selection')) initialState[action.payload] = ['#008000','#008000'];
      else initialState[action.payload] = '#3297FD';
      return initialState;
    case DEFAULT_COLORS:
      return initialState;
    default:
      return state;
  }
}

export default colorReducer;