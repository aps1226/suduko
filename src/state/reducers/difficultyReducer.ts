import { SET_DIFFICULTY } from '../actions/types';
const difficultyReducer = (state = null, action: any) =>{
  switch(action.type){
    case SET_DIFFICULTY:
      return action.payload;
    default:
      return state;
  }
}

export default difficultyReducer;