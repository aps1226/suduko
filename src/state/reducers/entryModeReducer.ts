import {SET_ENTRY_MODE} from '../actions/types';

const entryModeReducer = (state = true, action: any) =>{
  switch(action.type){
    case SET_ENTRY_MODE:
      return action.payload;
    default:
      return state;
  }
}

export default entryModeReducer;