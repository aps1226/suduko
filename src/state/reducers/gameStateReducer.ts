import { AnyAction } from 'redux';
import { SET_GAME_STATE } from '../actions/types';
import { GameState } from '../../../types';

const initGameState:GameState = {
  'isCompleted':false,
  'gameExists':false
}
const gameStateReducer = (state = initGameState, action: AnyAction):GameState =>{
  switch(action.type){
    case SET_GAME_STATE:
      return action.payload;
    default:
      return state;
  }
}

export default gameStateReducer;