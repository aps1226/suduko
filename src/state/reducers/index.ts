import {combineReducers} from 'redux';
import boardReducer from './boardReducer';
import colorReducer from './colorReducer';
import selectionReducer from './selectionReducer';

const reducers = combineReducers({
  board: boardReducer,
  colors: colorReducer,
  selection:selectionReducer
});

export default reducers;