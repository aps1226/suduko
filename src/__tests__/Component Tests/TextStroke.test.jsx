import React from 'react';
import { Text,View } from "react-native";
import { fireEvent, render, waitFor} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import {GameState} from '../../../types';
import {RootState} from '../../state/reducers/index';

import TextStroke from '../../../components/TextStroke';

//TextStroke component test.
describe('TextStroke', () =>{
  it('renders without crashing', () =>{
    const titleComponent = render(
      <TextStroke
        color = {'white'}
        stroke = {2}
      >
        <Text>
          Suduko
        </Text>
      </TextStroke>
    ).toJSON();
    expect(titleComponent).toMatchSnapshot();
  })
})