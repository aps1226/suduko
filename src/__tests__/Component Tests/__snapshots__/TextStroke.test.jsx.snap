// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextStroke renders without crashing 1`] = `
<View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": 0,
            "width": -2.4,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": 0,
            "width": 2,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": -2.4,
            "width": 0,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": -2,
            "width": -2,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": -2,
            "width": 2,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "textShadowColor": "white",
          "textShadowOffset": Object {
            "height": 2,
            "width": -2,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Suduko
    </Text>
  </View>
  <Text
    style={
      Object {
        "textShadowColor": "white",
        "textShadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "textShadowRadius": 1,
      }
    }
  >
    Suduko
  </Text>
</View>
`;
